### PR TITLE
Fix payment#payment_method for deleted records

### LINKED ISSUE #1186: Soft-deleted payment methods cause errors in admin
After deleting a payment method in the admin, any references to it on past orders will lead to 500 errors. (e.g.: The show page and all events that can be triggered on `/admin/orders/:number/payments`)

### PR DESCRIPTION
**Pull request by jhawthorn**
Fixes #1186

### CODE REVIEW

**Comment by adammathys** [comment]
:+1:

**Comment by mamhoff** [comment]
👍

File: core/app/models/spree/payment.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Payment < Spree::Base
 
     belongs_to :order, class_name: 'Spree::Order', touch: true, inverse_of: :payments
     belongs_to :source, polymorphic: true
-    belongs_to :payment_method, class_name: 'Spree::PaymentMethod', inverse_of: :payments
+    belongs_to :payment_method, -> { with_deleted }, class_name: 'Spree::PaymentMethod', inverse_of: :payments
```
Comment (jasonfb) [inline core/app/models/spree/payment.rb]: 👎

**Comment by jasonfb** [comment]
this is an anti-pattern and should not be allowed to proliferate throughout the codebase, although I do realize it fixes a bug today.

**Comment by mamhoff** [comment]
@jasonfb I agree with you. `acts_as_paranoid` is a bad replacement for properly time-windowed models. But as long as we don't have those, we need to work with what we have. I'm sorry.

**Comment by jasonfb** [comment]
Yes fair enough. This does indeed fix a bug so... "bob's your uncle." (That's something Europeans say right?)

**Comment by jhawthorn** [comment]
Wanted to note that I had to revert this because the new test caused a spurious failure :disappointed:

**Pull request merged**